RevenueCatUI/View+PresentPaywall.swift

```diff
@@ -560,14 +680,260 @@ private struct PresentingPaywallModifier: ViewModifier {
             self.restoreFailure?($0)
         }
         .interactiveDismissDisabled(self.purchaseHandler.actionInProgress)
+        .onAppear {
+            // Reset purchased flag so we can track if a purchase happens in THIS session.
+            // This is needed because PurchaseHandler is a @StateObject that persists.
+            self.purchaseHandler.resetSessionPurchaseResult()
```

Comment:
This is probably related to my earlier comment. But it's worth making sure that we're not calling `resetSessionPurchaseResult()` more than we need. I've seen `onAppear` be called more often than expected.

Reply:
Moving this to a `task` would do the trick right? 🤔

Reply:
That's always my first thought. But I'm not really sure. If we really want to be deterministic somehow, we should tie the task to some ID that we control and we know when it changes
```
.task(id: someID) {
    ...
}
```

We're doing that with `offering` in https://github.com/RevenueCat/purchases-ios/blob/7c12446fb3a56b5ca06db4858e24660d5f6b1167/RevenueCatUI/Templates/TemplateViewType.swift#L75-L77
but I don't have enough context to know what it should be tied to in this case